### PR TITLE
RX/TX Bar

### PR DESCRIPTION
**Pull request by spatializes**
#### What's this PR do?

Adds a status bar for displaying transmission events between the host and serial device. RX will flash blue when data is received from the connected device. TX will flash red when data is sent to the connected device.
#### Where should the reviewer start?

Pull this branch to see the new status bar below the debug output.
#### How should this be manually tested?

Connect a BASIC Stamp. Load in some looping code. Download code to the board. Notice the red flashing TX on download. Notice the blue flashing RX as debug information is sent to the host.
#### Any background context you want to provide?

Using flux with a new transmission store. Actions are dispatched from the device store when rx/tx events are found.
#### What are the relevant tickets?

References #78
#### Screenshots (if appropriate)

image

### CODE REVIEW

File: plugins/editor/index.js [hunk 1]
```diff
@@ -83,10 +87,17 @@ function editor(app, opts, done){
...
       outputConsole.style.backgroundColor = 'white';
       outputConsole.style.padding = '10px';
+      outputConsole.style.margin = '0';
       outputConsole.style.overflow = 'auto';
       outputConsole.style.whiteSpace = 'pre-wrap';
       el.appendChild(outputConsole);
     }
+    if(!transmission) {
+      transmission = document.createElement('div');
+      transmission.id = 'transmission';
```
Comment (phated) [inline plugins/editor/index.js]: you don't need to add an ID

File: plugins/editor/index.js [hunk 2]
```diff
@@ -83,10 +87,17 @@ function editor(app, opts, done){
...
+      el.appendChild(transmission);
+      React.render(<TransmissionBar />, document.getElementById('transmission'));
```
Comment (phated) [inline plugins/editor/index.js]: just pass the created `transmission` element, no reason to query for it

File: src/stores/transmission.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    };
+
+  }
+
+  onRx() {
+      this.setState({flashRx: true});
+
+      setTimeout(() => {
+        this.setState({ flashRx: false});
+      }, this.state.flashDuration);
```
Comment (phated) [inline src/stores/transmission.js]: `flashDuration` should be destructured

File: src/stores/transmission.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  onTx() {
+    this.setState({flashTx: true});
+
+    setTimeout(() => {
+      this.setState({ flashTx: false});
+    }, this.state.flashDuration);
```
Comment (phated) [inline src/stores/transmission.js]: `flashDuration` should be destructured

**Comment by spatializes** [comment]
updated based on comments above.

**Comment by spatializes** [comment]
updated with timeout changes

**Pull request merged**